keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,730 @@
+|                              |                    |                 |                      |                           |container_start_time_seconds                    |
+|                              |                    |                 |                      |                           |container_tasks_state                           |
+|                              |                    |                 |                      |                           |container_threads                               |
+|                              |                    |                 |                      |                           |container_threads_max                           |
+|                              |                    |                 |                      |                           |container_file_descriptors                      |
+|                              |                    |                 |                      |                           |container_last_seen                             |
+
+
+## Motivation
+
+We want to avoid using cAdvisor for container & pod level stats and move metric collection to the CRI implementation for the following reasons:
+
+* cAdvisor and metric dependency: CRI mission is not fully fulfilled - container runtime is not fully plugable 
+* It would be ideal to break the monolithic design of cAdvisor (i.e. cAdvisor currently needs to be aware of the underlying container runtimes) 
+* Duplicate stats are collected by both cAdvisor and the CRI runtime, which can lead to:
+	* Different information from different sources
+	* Confusion: Unclear origin of a given metric
+	* Performance degradations (increased CPU / Memory / etc) [xref][perf-issue]
+* Stats should be reported by the container runtime which knows behavior of the container/pod the best.
+* cAdvisor only supports runtimes that run processes on the host, not e.g. VM based runtime like Kata Containers.
+* cAdvisor only supports linux containers, not Windows ones.
+* cAdvisor has big list of external dependencies since it needs to support multiple container runtimes itself - leads to large support surface.
+
+[perf-issue]: https://github.com/kubernetes/kubernetes/issues/51798
+
+
+### Goals
+
+* Rely on a single component to provide the container and pod level metrics for the Summary API.
+* Improve performance and reduce confusion on metrics collection in the Kubelet.
+* Do not introduce breaking changes to the Summary API.
+* Eliminate dependencies on container runtime clients used by cAdvisor.
+* Move or deprecate the stats exclusively reported by `/metrics/cadivsor` to be collected (and maybe reported) by the CRI implementation.
+
+### Non-Goals
+
+- Have CRI support Volume Plugin stats (responsibility of the Kubelet).
+- Have CRI support Ephemeral Storage stats (responsibility of the Kubelet).
+- Have CRI support metrics of the rootfs (responsibility of cAdvisor).
```

Comment:
This contradicts the table above, which says this is the responsibility of the CRI or N/A. Was one of these a typo?

Reply:
rootfs as in the node host fs, it should be clearer